server/apps/packages/package_service.py

```diff
@@ -222,6 +222,50 @@ def get_packages(self, doc_id):
 
         return get_resource_service(ARCHIVE).get_from_mongo(req=request, lookup=query)
 
+    def remove_ref_from_inmem_package(self, package, ref_id):
+        """
+        Removes the reference with ref_id from non-root groups. If there is nothing left
+        in that group then the group and its reference in root group is also removed.
+        If the removed item was the last item then returns
+        :param package: Package
+        :param ref_id: Id of the reference to be removed
+        :return: True if there are still references in the package, False otherwise
+        """
+        groups_to_be_removed = []
```

Comment:
The number of items here won't be that many so I'll keep it like this for the time being, thanks for pointing out.